magenta-lib/src/main/scala/magenta/tasks/ChangeSetTasks.scala

```diff
@@ -0,0 +1,119 @@
+    )
+
+    val (stackName, changeSetType) = stackLookup.lookup(reporter, cfnClient)
+
+    val template = processTemplate(stackName, templateString, s3Client, stsClient, region, reporter)
+    val parameters = unresolvedParameters.resolve(template, accountNumber, changeSetType, reporter, cfnClient)
+
+    reporter.info("Creating Cloudformation change set")
+    reporter.info(s"Stack name: $stackName")
+    reporter.info(s"Change set name: ${unresolvedParameters.changeSetName}")
+    reporter.info(s"Parameters: $parameters")
+
+    CloudFormation.createChangeSet(reporter, unresolvedParameters.changeSetName, changeSetType, stackName, unresolvedParameters.stackTags, template, parameters, cfnClient)
+  }
+
+  def description = s"Create change set ${unresolvedParameters.changeSetName} for stack ${stackLookup.strategy} with ${templatePath.fileName}"
+  def verbose = description
+}
+
+class CheckChangeSetCreatedTask(
+  region: Region,
+  stackLookup: CloudFormationStackLookup,
+  override val duration: Long
+)(implicit val keyRing: KeyRing, artifactClient: AmazonS3) extends Task with RepeatedPollingCheck {
+
+  override def execute(reporter: DeployReporter, stopFlag: => Boolean): Unit = {
+    check(reporter, stopFlag) {
+      val cfnClient = CloudFormation.makeCfnClient(keyRing, region)
+
+      val (stackName, _) = stackLookup.lookup(reporter, cfnClient)
+      val changeSetName = stackLookup.changeSetName
+
+      val request = new DescribeChangeSetRequest().withChangeSetName(changeSetName).withStackName(stackName)
+      val response = cfnClient.describeChangeSet(request)
+
+      shouldStopWaiting(response.getStatus, response.getStatusReason, response.getChanges.asScala, reporter)
+    }
+  }
+
+  def shouldStopWaiting(status: String, statusReason: String, changes: Iterable[Change], reporter: DeployReporter): Boolean = status match {
+    case "CREATE_COMPLETE" => true
+    case "FAILED" if changes.isEmpty => true
+    case "FAILED" => reporter.fail(statusReason)
+    case _ =>
+      reporter.verbose(status)
+      false
+  }
+
+  def description = s"Checking change set ${stackLookup.changeSetName} creation for stack ${stackLookup.strategy}"
+  def verbose = description
+}
+
+class ExecuteChangeSetTask(
+  region: Region,
+  stackLookup: CloudFormationStackLookup,
+)(implicit val keyRing: KeyRing, artifactClient: AmazonS3) extends Task {
+  override def execute(reporter: DeployReporter, stopFlag: => Boolean): Unit = {
+    val cfnClient = CloudFormation.makeCfnClient(keyRing, region)
+
+    val (stackName, _) = stackLookup.lookup(reporter, cfnClient)
+    val changeSetName = stackLookup.changeSetName
+
+    val describeRequest = new DescribeChangeSetRequest().withChangeSetName(changeSetName).withStackName(stackName)
+    val describeResponse = cfnClient.describeChangeSet(describeRequest)
+
+    if(describeResponse.getChanges.isEmpty) {
+      reporter.info(s"No changes to perform for $changeSetName on stack $stackName")
+    } else {
```

Comment:
I would be interested in seeing the changes about to be made here via a `verbose` report? I imagine it might be quite verbose but I think this would be super useful feedback, especially if it goes wrong for some reason.